### PR TITLE
Fix error linking (set build pugixml as shared lib)

### PR DESCRIPTION
**Pull request by baytuch**

### CODE REVIEW

File: RenderSystems/GLSupport/CMakeLists.txt [hunk 1]
```diff
@@ -94,7 +94,7 @@ if(OGRE_CONFIG_ENABLE_GL_STATE_CACHE_SUPPORT)
     PROPERTIES COMPILE_DEFINITIONS OGRE_ENABLE_STATE_CACHE)
 endif()
 
-add_library(OgreGLSupport STATIC 
+add_library(OgreGLSupport SHARED
```
Comment (paroj) [inline RenderSystems/GLSupport/CMakeLists.txt]: this unrelated change breaks all builds, please revert

Comment (Lastick) [reply RenderSystems/GLSupport/CMakeLists.txt]: I did not check the fix in windows, but I saw a compiler bot report. Apparently, this is indeed so. But I was embarrassed when I compiled into Linux that this library was static, when like the others were - shared. I will definitely study the issue in more detail.

Comment (baytuch) [reply RenderSystems/GLSupport/CMakeLists.txt]: I did not check the fix in windows, but I saw a compiler bot report. Apparently, this is indeed so. But I was embarrassed when I compiled into Linux that this library was static, when like the others were - shared. I will definitely study the issue in more detail.

Comment (paroj) [reply RenderSystems/GLSupport/CMakeLists.txt]: it is intentionally static, so the rendersystem plugins are self-contained. Allowing it shared would require more CMake changes for proper GL/ EGL dependency handling.

**Comment by paroj** [comment]
should be fixed in master
